[RISCV] Move RISCVVType namespace to TargetParser

wangpc-pp opened the pull request:
Clang and some middle-end optimizations may need these helper
functions.

This can reduce some duplications.

topperc commented:
Is it possible to move them to TargetParser instead?

wangpc-pp commented:
> Is it possible to move them to TargetParser instead?

Yes, I think we can. But is there any specific reason?

topperc commented:
> > Is it possible to move them to TargetParser instead?
> 
> Yes, I think we can. But is there any specific reason?

The Support library shouldn't contain target specific code if possible. The only reason RISCVISAInfo is in there is because TableGen uses it.

wangpc-pp committed:
[RISCV] Move RISCVVType namespace to TargetParser

Clang and some middle-end optimizations may need these helper
functions.

This can reduce some duplications.

wangpc-pp commented:
Ping.

topperc commented on llvm/unittests/TargetParser/CMakeLists.txt:
```diff
@@ -8,6 +8,7 @@ add_llvm_unittest(TargetParserTests
   Host.cpp
   TargetParserTest.cpp
   TripleTest.cpp
+  RISCVTargetParserTest.cpp
```
alphabetize

wangpc-pp committed:
Alphabetize tests

topperc reviewed (approved):
LGTM

wangpc-pp commented:
There are some failures: https://lab.llvm.org/buildbot/#/builders/57/builds/33206
Fix is here: https://github.com/llvm/llvm-project/pull/84109
